components/Favorites.tsx

```diff
@@ -1,74 +1,103 @@
-import React, { useState, useEffect } from 'react';
-import Book from './Book';
-import styles from '../styles/book-search.module.css';
-import { languages, toggleStringInList } from './BookSearch';
+import React, { useEffect, useState } from "react";
 
-function Favorites({ favorites, handleFavorite }) {
-  const [favoriteBooks, setFavoriteBooks] = useState([]);
-  const [searchQuery, setSearchQuery] = useState('');
-  const [filterTags, setFilterdTags] = useState([]);
+import styles from "../styles/book-search.module.css";
+import { Book } from "./Book";
+import { languages, toggleStringInList } from "./BookSearch";
+import type { AppContextType, Book_t } from "./appContext";
+
+function Favorites({ favorites, handleFavorite }: AppContextType) {
+  const [favoriteBooks, setFavoriteBooks] = useState<Book_t[]>([]);
+  const [searchQuery, setSearchQuery] = useState("");
+  const [filterTags, setFilterdTags] = useState<string[]>([]);
   const [isFirstLoad, setIsFirstLoad] = useState(true);
 
   useEffect(() => {
-
     setIsFirstLoad(false);
 
-    const savedFavorites = JSON.parse(localStorage.getItem('favorites')) || favorites;
+    const savedFavoritesJSON = localStorage.getItem("favorites");
+    const savedFavorites =
+      savedFavoritesJSON !== null
+        ? (JSON.parse(savedFavoritesJSON) as Book_t[])
+        : [];
```

Comment:
czemu ładujesz to na nowo skoro masz z props'ów tablice favorites? nie mogłeś jej użyć?